src/Domain/DomainHelpers.hpp

```diff
@@ -65,6 +65,20 @@ void set_periodic_boundaries(
         std::unordered_map<Direction<VolumeDim>, BlockNeighbor<VolumeDim>>>*>
         neighbors_of_all_blocks);
 
+/// \ingroup ComputationalDomainGroup
+/// \brief The ordered graph for a domain made of layers of Blocks.
+///
+/// The `domain_extents` argument holds the number of blocks to have
+/// in each dimension. The blocks all have aligned orientations by
+/// construction. The `block_ids_to_exclude` argument allows to user
+/// to selectively exclude blocks from the resulting domain. This allows
+/// for the creation of non-trivial shapes such as the net for a tesseract.
+template <size_t VolumeDim>
```

Comment:
how do you know the block_id's to exclude?  Instead of a single number, let's specify a multi-index (i.e. an Index<VolumeDim> into the domain_extents for each block to exclude